src/Workspaces/Core/Portable/CodeGeneration/Symbols/CodeGenerationAbstractMethodSymbol.cs

```diff
@@ -71,6 +70,10 @@ public override TResult Accept<TResult>(SymbolVisitor<TResult> visitor)
 
         public override SymbolKind Kind => SymbolKind.Method;
 
+        public virtual bool IsDeclaredReadOnly => throw new NotImplementedException();
+
+        public virtual bool IsEffectivelyReadOnly => throw new NotImplementedException();
```

Comment:
Are derived types supposed to implement these properties? If so, consider making these `abstract`. If not, consider removing `virtual`.

Reply:
There's two types that require concrete implementations for this.

1. `CodeGenerationConstructedMethodSymbol` might implement them by delegating to `ConstructedFrom`
2. `CodeGenerationMethodSymbol` has a `ContainingType` and `Modifiers` but it seems like implementing them essentially involves copy-pasting.

I'd like to know more about how these types are used before attempting to really implement these properties on them. For the moment I can do the "obvious" thing for (1) and kick the implementation problem over to (2).

Reply:
Perhaps @CyrusNajmabadi has some guidance here?

Reply:
Finished the change described in (1) and throwing NotImplementedException for (2) until we get some clearer sense about the need for a real implementation.